Remove quarantine functionality

replaceafill opened the pull request:
This PR removes the code related to the quarantine feature.

I've recorded how the workflow looks in `qa/1.x` with the quarantine feature and how it looks now in this https://github.com/artefactual/archivematica/issues/598#issuecomment-567388848 and added links to SVG files which hopefully will make reviewing this easier.

The initial commit in this PR was rebased from @sevein's branch: https://github.com/artefactual/archivematica/tree/dev/issue-598-remove-quarantine

Connected to https://github.com/artefactual/archivematica/issues/598

ross-spencer commented on src/MCPServer/lib/server/jobs/decisions.py:
```diff
@@ -119,11 +119,8 @@ def decide(self, choice):
 
-        # TODO: the default automated processing config uses a choice that isn't
-        # listed in the workflow (d4404ab1-dc7f-4e9e-b1f8-aa861e766b8e, for link
-        # 755b4177-c587-41a7-8c52-015277568302). Fix that.
```
Link gone, comment gone! :+1:

ross-spencer reviewed (approved):
Great work both, very happy to approve this one. :+1:

replaceafill committed:
Remove the quarantine functionality

This removes the workflow code related to the quarantine feature and
updates the processing configurations and watched directories.

The workflow paths were updated to:

* All bag transfers: Assign UUIDs to directories -> Scan for Viruses

* Dspace: Verify metadata directory checksums -> Scan for Viruses

* Dataverse and Standard transfer: Verify metadata directory checksums
  -> Scan for Viruses

* TRIM: Assign UUIDs to directories -> Scan for Viruses

Co-authored-by: Jesús García Crespo <jesus@sevein.com>
